### PR TITLE
KAZOO-4033 handle bypass_media

### PR DESCRIPTION
**Pull request by lazedo**

### CODE REVIEW

File: applications/ecallmgr/src/ecallmgr_fs_bridge.erl [hunk 1]
```diff
@@ -120,40 +123,48 @@ handle_hold_media(DP, _Node, UUID, JObj) ->
...
--spec maybe_bypass_endpoint_media(wh_json:objects(), wh_proplist()) -> wh_proplist().
-maybe_bypass_endpoint_media([Endpoint], DP) ->
-    case wh_json:is_true(<<"Bypass-Media">>, Endpoint) of
-        'true' ->
-            [{"application", "set bypass_media=true"}|DP];
-        'false' ->
-            DP
+-spec maybe_bypass_endpoint_media(wh_json:objects(), ne_binary(), ne_binary(), wh_proplist()) -> wh_proplist().
+maybe_bypass_endpoint_media([Endpoint], BridgeProfile, ChannelProfile, DP) ->
+    EndpointProfile = wh_json:get_value(<<"SIP-Interface">>, Endpoint, BridgeProfile),
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_fs_bridge.erl]: Use `get_ne_binary_value` ?

File: applications/ecallmgr/src/ecallmgr_fs_bridge.erl [hunk 2]
```diff
@@ -178,15 +189,29 @@ pre_exec(DP, _Node, _UUID, _JObj) ->
      |DP
     ].
 
--spec create_command(wh_proplist(), atom(), ne_binary(), wh_json:object()) -> wh_proplist().
-create_command(DP, _Node, _UUID, JObj) ->
-    Endpoints = wh_json:get_ne_value(<<"Endpoints">>, JObj, []),
+-spec create_command(wh_proplist(), atom(), ne_binary(), channel(), wh_json:object()) -> wh_proplist().
+create_command(DP, _Node, _UUID, #channel{profile=ChannelProfile}, JObj) ->
+    BypassAfterBridge = ?BYPASS_MEDIA_AFTER_BRIDGE,
+    BridgeProfile = wh_util:to_binary(wh_json:get_value(<<"SIP-Interface">>, JObj, ?DEFAULT_FS_PROFILE)),
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_fs_bridge.erl]: Same here

File: applications/ecallmgr/src/ecallmgr_fs_bridge.erl (hunk 1 above)
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_bridge.erl]: no need

File: applications/ecallmgr/src/ecallmgr_fs_bridge.erl [hunk 3]
```diff
@@ -178,15 +189,29 @@ pre_exec(DP, _Node, _UUID, _JObj) ->
...
                                 ,try_create_bridge_string(Endpoints, JObj)
                                ]),
     [{"application", BridgeCmd}|DP].
 
+-spec maybe_bypass_after_bridge(boolean(), ne_binary(), ne_binary(), wh_json:objects()) -> wh_json:objects().
+maybe_bypass_after_bridge('false', _, _, Endpoints) ->
+    [wh_json:delete_key(<<"Bypass-Media">>, Endpoint) || Endpoint <- Endpoints];
+maybe_bypass_after_bridge('true', BridgeProfile, ChannelProfile, Endpoints) ->
+    [begin
+         case wh_json:get_value(<<"SIP-Interface">>, Endpoint, BridgeProfile) of
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_fs_bridge.erl]: Again

File: applications/ecallmgr/src/ecallmgr_fs_bridge.erl (hunk 2 above)
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_bridge.erl]: no need

File: applications/ecallmgr/src/ecallmgr_fs_bridge.erl (hunk 3 above)
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_bridge.erl]: no need again

**Pull request merged**